core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/workflow/impl/IntegerProgressStep.java

```diff
@@ -0,0 +1,41 @@
+package uk.ac.bbsrc.tgac.miso.core.data.workflow.impl;
+
+import uk.ac.bbsrc.tgac.miso.core.data.workflow.AbstractProgressStep;
+import uk.ac.bbsrc.tgac.miso.core.data.workflow.WorkflowStep;
+
+public class IntegerProgressStep extends AbstractProgressStep {
+  private static final long serialVersionUID = 1L;
+
+  private int input;
+
+  public int getInput() {
+    return input;
+  }
+
+  public void setInput(int input) {
+    this.input = input;
+  }
+
+  @Override
+  public int hashCode() {
+    final int prime = 31;
+    int result = super.hashCode();
+    result = prime * result + input;
+    return result;
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (this == obj) return true;
+    if (!super.equals(obj)) return false;
+    if (getClass() != obj.getClass()) return false;
+    IntegerProgressStep other = (IntegerProgressStep) obj;
+    if (input != other.input) return false;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this if-then-else statement by a single return statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1126)